### PR TITLE
[LogRhythmRest v2] - fix issue with uploading non-text files

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-30366)

## Description
Fixes an issue where it was not possible to upload into a case non-text files (e.g.: binary files of any kind)

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py [hunk 1]
```diff
@@ -2441,6 +2453,7 @@ def fetch_incidents_command(client: Client, fetch_type: str, cases_max_fetch: in
     elif fetch_type == 'Cases':
         return fetch_cases(client, cases_max_fetch, fetch_time, fetch_case_evidences,
                            case_tags_filter, case_status_filter, case_priority_filter)
+    return None
```
Comment (GuyAfik) [inline Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py:2456]: this was added by pre-commit

**Review by sapirshuker — APPROVED** [review]
Nice!!

File: Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py [hunk 2]
```diff
@@ -1207,17 +1207,29 @@ def case_file_evidence_add_request(self, case_id, entry_id):
...
-               '-------------------------------'
-
-        response = self._http_request('POST', f'lr-case-api/cases/{case_id}/evidence/file', data=data)
+        content_type = mimetypes.guess_type(file_name)[0]
+        demisto.debug(f'content-type of {file_name} is {content_type=}')
+
+        try:
+            data = '-----------------------------\n' \
+                   f'Content-Disposition: form-data; name="file"; filename="{file_name}"\n' \
+                   f'Content-Type: {content_type}\n\n' \
```
Comment (sapirshuker) [inline Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py:1216]: Try to add it to a function, in order to avoid duplicate code

Comment (GuyAfik) [reply Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py:1216]: good idea, will do

File: Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py (hunk 1 above)
Comment (JudahSchwartz) [reply Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py:2452]: which hook? @ilaner

File: Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py [hunk 3]
```diff
@@ -1207,17 +1217,15 @@ def case_file_evidence_add_request(self, case_id, entry_id):
...
-               '-----------------------------\n' \
-               'Content-Disposition: form-data; name="note"\n\n' \
-               '-------------------------------'
+        content_type = mimetypes.guess_type(file_name)[0]
+        demisto.debug(f'content-type of {file_name} is {content_type}')
 
-        response = self._http_request('POST', f'lr-case-api/cases/{case_id}/evidence/file', data=data)
+        try:
+            response = self._http_request('POST', f'lr-case-api/cases/{case_id}/evidence/file', data=prepare_data(content_type))
+        except DemistoException as error:
```
Comment (JudahSchwartz) [inline Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py:1225]: theres no better way to detect this before the first call?

Comment (GuyAfik) [reply Packs/LogRhythmRest/Integrations/LogRhythmRestV2/LogRhythmRestV2.py:1225]: You can detect it by file extensions maybe, but could be buggy IMO.

**Pull request closed without merging**